src/Grid/mixed_grid.jl

```diff
@@ -0,0 +1,51 @@
+mutable struct MixedGrid{dim, C, T<:Real} <: Ferrite.AbstractGrid{dim}
+    cells::C # Tuple of concretely typed cell vectors
+    ncells_per_vector::Vector{Int}
```

Comment:
Is `ncells_per_vector` needed? 
Did some quick benchmarks for `getncells` and couldn't see that it would be faster (actually slower but so fast that it wouldn't matter anyway)